File: python/num_stud_ute.py
def unable_to_eat():
    students = [0,1,0,1]
    sandwiches = [1,0,1,1]
    # Basic format. Change to allow for user input.

    total = len(students)
    top = sandwiches[0]
    eat = 0

    while (students.count(top) > 0):
        if (students[0] == top):
            eat = eat + 1
            students.pop(0)
            sandwiches.pop(0)
        else:
            students.append(students.pop(0))
        
        if (len(sandwiches) > 0):
            top = sandwiches[0]

        else:
            break
    return int(total - eat)

File: java/NumStudUte.java
public class NumStudUte {
    private static NumStudUte instance;

    public static NumStudUte getInstance(){
        if (instance == null){
            instance = new NumStudUte();
        }
        return instance;
    }

    private NumStudUte(){

    }

    public void run(){
        System.out.println("Determining the number of studentsunable to eat:");
        
    }

    public int determineEatCount(){
        int students[] = {0, 1, 0, 1};
        int sandwiches[] = {1, 0, 1, 0};

        int total = students.length;
        int top = sandwiches[0];
        int eat = 0;

        return (total - eat);
    }
}
